src/cr/cube/dimension.py

```diff
@@ -725,6 +734,214 @@ def _elements(self):
         return tuple(element for element in self._all_elements if not element.missing)
 
 
+class _DimensionShimElementIds(object):
+    """Object used to replace element ids with alias for subvariables
+
+    We want to move to a world where elements on a subvariables dimension are
+    identified by their alias, but right now the "element_id" from zz9 is
+    an index, and the transforms have several different ways to refer to
+    subvariables.
+
+    Types of identifiers for subvariables (and derived insertions):
+    - "element_id": Stored in the cube result as the object name in
+        `dimensions[i].type.elements[j].id`. For subvariables, zz9 currently puts
+        the index integer here. Long term zz9 may change this to the the alias.
+    - "subvariable_id": Subvariables have an id stored in
+        `dimensions[i].type.elements[j].value.id`, generally this is a 4 digit,
+        0-padded index of the subvariable when it was first created (eg "0001",
+        "0002", ...), though it is not required to be. For derived insertions,
+        currently the name is used here.
+    - "alias": Subvariables also have an alias that identifies them. It
+    """
+
+    def __init__(self, dimension_type, dimension_dict, dimension_transforms_dict):
+        self._dimension_type = dimension_type
+        self._dimension_dict = dimension_dict
+        self._dimension_transforms_dict = dimension_transforms_dict
+
+    @lazyproperty
+    def shimmed_dimension_dict(self):
+        """Copy of dimension dictionary with shimmed `element_id`s
+
+        We want to move to a world where elements on a subvariables dimension are
+        identified by their alias, but right now the "element_id" from zz9 is
+        an index for subvariables.
```

Comment:
I keep catching on "is an index". I guess it's not a term that commonly has meaning to me when appearing on it's own. Maybe `(unstable) element-index` or `is simply the subvariable's (unstable) cardinal position in the subvariable sequence`.